x/concentrated-liquidity/tick.go

```diff
@@ -400,22 +408,40 @@ func (k Keeper) GetTickLiquidityNetInDirection(ctx sdk.Context, poolId uint64, t
 	for ; iterator.Valid(); iterator.Next() {
 		tickIndex, err := types.TickIndexFromBytes(iterator.Key())
 		if err != nil {
-			return []queryproto.TickLiquidityNet{}, err
+			return []queryproto.TickLiquidityNet{}, 0, sdk.Dec{}, err
 		}
 
 		tickStruct, err := ParseTickFromBz(iterator.Value())
 		if err != nil {
-			return []queryproto.TickLiquidityNet{}, err
+			return []queryproto.TickLiquidityNet{}, 0, sdk.Dec{}, err
 		}
 
 		liquidityDepth := queryproto.TickLiquidityNet{
 			LiquidityNet: tickStruct.LiquidityNet,
 			TickIndex:    tickIndex,
 		}
+
+		// If we are using the current tick, there is no need to accumulate liquidity values here since we store current tick liquidity in state.
+		if startTick == p.GetCurrentTick() {
+			liquidityDepths = append(liquidityDepths, liquidityDepth)
+			continue
+		}
```

Comment:
I don't believe this is correct in zero for one direction, only in one for zero.

In zero for one (left direction), we want to be including the current tick liquidity because "current tick" equal to start tick implies that we still need to cross that tick when going left. This is due to our "active range" definition that is inclusive of the lower tick but not upper.

I think that the correct approach to handle this edge case is in this PR:https://github.com/osmosis-labs/osmosis/pull/5578 where we bump the `startTick` value by one for the iterator.

If zero for one, we use reverse iterator. As a result, we need to increment the start tick by 1
so that we include the start tick in the search.

If one for zero, we use a forward iterator. However, our definition of the active range is inclusive
of the lower bound. As a result, current liquidity must already include the lower bound tick
so we skip it.

Could you please review #5578? Once we agree on approach, we can drive both PRs to completion

Reply:
> In one for zero (right direction), the current tick equal to the start tick implies that we have already crossed that tick so we don't need to return its depth.

I don't fully understand this claim, isn't the purpose of this query to return liquidity depth of a tick? It should be up to the client to decide whether they use the current tick or not depending on which direction they are swapping. 

I think the answer might differ depending on the purpose of the query, but as of my understanding rn, I dpm't think this is wrong? 🤔